### DISCUSSION TITLE
What is the "| quote" syntax in a template?

### DISCUSSION THREAD

**Discussion by PaulRBerg** [twpayne/chezmoi#2109] [Q&A] [answered]
I was reading the [Manage machine-to-machine differences](https://www.chezmoi.io/user-guide/manage-machine-to-machine-differences/#use-templates) guide, when I noticed this syntax:

```mustache
{{ .email | quote }}
```

I gather that this is a [mustache](https://mustache.github.io/)-like templating system, but I'm not quite sure what is the `| quote` syntax doing?

Does it simply add quotes to the value?

**Comment by twpayne** [comment, marked as answer]
As described in the [templating guide](https://www.chezmoi.io/user-guide/templating/), chezmoi uses Go's [`text/template`](https://pkg.go.dev/text/template). The `| quote` takes the value of what's on the left hand side of the `|` and quotes it as a string, including escaping any special characters like `\` and `"`.

You could use

    "{{ .email }}"

but, depending on the exact context, this will break if `.email` contains any backslashes, double quotes, or newlines.

**Reply by PaulRBerg** [reply]
Thanks! And sorry, I missed the `Templating` guide.
